### PR TITLE
Greengrocers React - Marian

### PR DESCRIPTION
**Pull request by marianphillips**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -2,60 +2,124 @@ import './styles/reset.css'
...
+  const [cartItems, setCartItems] = useState([])
+  const [showingDetail, setShowingDetail] = useState(false)
+  const [detailItem, setDetailItem] = useState({})
+  const [purchasePage, setPurchasePage] = useState(false)
+  const [storeOrStock, setStoreOrStock] = useState('store')
 
-What should a cart item look like? 🤔
-*/
+  const alreadyInCart = item => {
+    return cartItems.find(cartItem => cartItem.name === item.name) !== undefined
```
Comment (alsyoufomar) [inline src/App.js:23]: you could have used some() method, it returns a boolean

File: src/App.js [hunk 2]
```diff
@@ -1,61 +1,59 @@
...
+const [detailItem, setDetailItem] = useState({})
+
+const alreadyInCart = (item) => {
+  return cartItems.find(cartItem => cartItem.name === item.name) !== undefined
+
+}
+
+const addToCart = (item) => {
+    if(alreadyInCart(item)) {
+  const updatedCart = cartItems.map(cartItem => cartItem.name === item.name ? {...cartItem, quantity: cartItem.quantity+= 1} : cartItem)
```
Comment (nicopicchio) [inline src/App.js]: I like the use of the ternary operator. I need to understand it better so I can start using it but I am still debating how it looks from a readability point of view.

File: src/Header.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    if (filterValue === "lowest-price") {
+      filteredStoreItems.sort((a, b) => a.price - b.price);
+    }
+  
+    if (filterValue === "highest-price") {
+      filteredStoreItems.sort((a, b) => b.price - a.price);
+    }
+  
+    if (filterValue === "alphabetically") {
+      filteredStoreItems.sort((a, b) => a.name.localeCompare(b.name));
```
Comment (nicopicchio) [inline src/Header.js:25]: Need to look into this localeCompare, never used it before and not sure what it does.

File: src/Cart.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+export default function Cart(props) {
```
Comment (alsyoufomar) [inline src/Cart.js:1]: oioioi that's awesome!

**Pull request closed without merging**